### ISSUE TITLE
Cannot craft if the output is the same as the input.

### ISSUE THREAD

**Issue by kalutro** [AppliedEnergistics/Applied-Energistics-2#7148]
### Describe the bug

Created a pattern that inputs 6 Iron Ingots and 1 Universal Press (LazierAE2) and outputs 2 Universal Press. I have the required materials, but the crafting screen will not let me craft. The error is: "Partial Plan: Missing Ingredients" even though it shows I have the required ingredients. I believe this may be because the output is the same as the input.

![javaw_WzSHnoFubo](https://github.com/AppliedEnergistics/Applied-Energistics-2/assets/69569507/11b4fe4b-57af-4f6c-8e42-17f53aafefad)

Note that if I try to craft two Universal Press, I get the expected result showing I do not have enough materials.

### How to reproduce the bug

Requires LazierAE2 (maybe).

Create a pattern for the Universal Press (6 Iron Ingots and 1 Universal Press).
Place in Pattern Provider connected to a Circuit Etcher (LazierAE2).
Attempt to craft a Universal Press.

### Expected behavior

I would expect to be able to complete the craft.

### Additional details

N/A

### Which minecraft version are you using?

1.18

### On which mod loaders does it happen?

Forge

### Crash log

N/A

**Comment by DAmNRelentless** [comment]
It is because the input is the same as the output, yes. Recursive recipes are not supported in general. You shouldn't need so many universal presses anyways that it's worth automating. So you either duplicate them manually or you need to use a normal inscriber where you leave the universal press in and only encode the block of iron in the recipe so the inscriber keeps cloning the press. It's a bit more expensive but this way you can automate the process if you really need to.

**Comment by kalutro** [comment]
I guess the solution would be to provide a more specific error message as the current one makes no sense. Something like, "Recursive crafting not allowed."

**Comment by Technici4n** [comment]
Yes, that's a good idea and would solve these issues that keep being opened. Might even be nice to make the pattern invalid.
